code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -2471,39 +2471,17 @@
 		return 1
 
 	switch(volume)
-		if(1 to 5)
+		if(0.1 to 5)
 			if(ishuman(M))
 				var/mob/living/carbon/human/H = M
 				if(H.species.name != "Diona")
-					var/datum/organ/external/affecting = H.get_organ()
-					for(var/datum/wound/W in affecting.wounds)
-						spawn(1)
-							affecting.wounds -= W
-							H.visible_message("<span class='warning'>[H]'s wounds close up in the blink of an eye!</span>")
-					if(H.getOxyLoss()>0 && prob(90))
-						if(holder.has_reagent("mednanobots"))
-							H.adjustOxyLoss(-4)
-							holder.remove_reagent("mednanobots", 4/40)  //The number/40 means that every time it heals, it uses up number/40ths of a unit, meaning each unit heals 40 damage
-						else
-					if(H.getBruteLoss()>0 && prob(90))
-						if(holder.has_reagent("mednanobots"))
-							H.heal_organ_damage(5, 0)
-							holder.remove_reagent("mednanobots", 5/40)
-						else
-					if(H.getFireLoss()>0 && prob(90))
-						if(holder.has_reagent("mednanobots"))
-							H.heal_organ_damage(0, 5)
-							holder.remove_reagent("mednanobots", 5/40)
-						else
-					if(H.getToxLoss()>0 && prob(50))
-						if(holder.has_reagent("mednanobots"))
-							H.adjustToxLoss(-2)
-							holder.remove_reagent("mednanobots", 2/40)
-						else
-					if(H.getCloneLoss()>0 && prob(60))
+					if(H.getOxyLoss()>0 || H.getBruteLoss()>0 || H.getToxLoss()>0 || H.getFireLoss()>0 || H.getCloneLoss()>0)
 						if(holder.has_reagent("mednanobots"))
-							H.adjustCloneLoss(-2)
-							holder.remove_reagent("mednanobots", 2/40)
+							H.adjustOxyLoss(-5)
+							H.heal_organ_damage(5, 5)
+							H.adjustToxLoss(-5)
+							H.adjustCloneLoss(-5)
+							holder.remove_reagent("mednanobots", 10/40)  //The number/40 means that every time it heals, it uses up number/40ths of a unit, meaning each unit heals 40 damage
```

Comment:
Yes, my point was to make Mednanos more consistent in that it would heal the the same regardless of what kind of damage you took. Before it would rely on an RNG system and would be largely ineffective on Toxin/Oxygen damage. This makes it so that it is effective regardless of what damage is caused.